docs/guides/02_writing_commands.md

```diff
@@ -0,0 +1,233 @@
+
+```csharp
+public class ProjectsRepository<TContext>
+    : EFRepository<Purchase, SId<Purchase>, TContext>
+    where TContext : DbContext
+{
+    public ProjectsRepository(TContext dbContext)
+        : base(dbContext) { }
+
+    public override Task<Purchase?> FindAsync(SId<Purchase> id, CancellationToken cancellationToken = default)
+    {
+        return DbSet.AsTracking().FirstOrDefaultAsync(p => p.Id == id, cancellationToken)!;
+    }
+}
+```
+
+Here `EFRepository` is a class from Core Library containing default implementations for `Add`, `Delete` and `Update` methods which can be used to alter the contents of the database. Usually `FindAsync` method can be implemented as a simple retrieval of an entity based on Id but more complicated operations can also be done in the method if they are needed.
+
+> **Warning :warning:**
+> If an implemented command does not work, make sure that all referenced types are properly registered in the dependency injection container.
+
+## More complex commands
+
+The command for creating a project is a really basic one - it only takes a name for new project and all of the validation can be done without checking the state of the domain. We will now show a slightly more complex command which will be responsible for adding tasks to existing projects.
+
+```csharp
+[AllowUnauthorized]
+public class AddTasksToProject : ICommand
+{
+    public string ProjectId { get; set; }
+    public List<TaskDTO> Tasks { get; set; }
+
+    public static class ErrorCodes
+    {
+        public const int ProjectDoesNotExist = 1;
+        public const int TasksCannotBeNull = 2;
+        public const int TasksCannotBeEmpty = 3;
+    }
+}
+
+
+public class TaskDTO
+{
+    public string Name {get; set; }
+
+    public class ErrorCodes
+    {
+        public const int NameCannotBeEmpty = 101;
+        public const int NameTooLong = 102; 
+    }
+}
+```
+
+Notice the use of `TaskDTO` class. DTO classes can be defined as part of contracts and matching classes will be generated for web and mobile clients. Also, notice that in `TaskDTO` we used error codes starting from 101 to avoid clashes with error codes in `AddTasksToProject`. Now, let's move onto validating the command:
+
+```csharp
+public class AddTasksToProjectCV : ContextualValidator<AddTasksToProject>
+{
+    public AddTasksToProjectCV()
+    {
+        RuleFor(cmd => cmd.Tasks)
+            .NotNull()
+            .WithCode(AddTasksToProject.ErrorCodes.TasksCannotBeNull)
+            .NotEmpty()
+            .WithCode(AddTasksToProject.ErrorCodes.TasksCannotBeEmpty);
+
+        RuleForEach(cmd => cmd.Tasks)
+            .ChildRules(child => child.RuleFor(c => c).SetValidator(new TaskDTOValidator()));
+
+        RuleForAsync(cmd => cmd, CheckProjectExistsAsync)
```

Comment:
Okay, this will be "slight" breaking change here - instead of `RuleForAsync`, let's use `MustAsync`. It will be more natural and should be _the_ way of validating a single thing.

Also, `RuleForAsync` will probably be deprecated soon, as we don't want to stay on old version of FV (and we can't implement that in newer version).